codec-http/src/main/java/io/netty/handler/codec/http/HttpRequestDecoder.java

```diff
@@ -301,4 +301,15 @@ protected HttpMessage createInvalidMessage() {
     protected boolean isDecodingRequest() {
         return true;
     }
+
+    @Override
+    protected boolean isContentAlwaysEmpty(final HttpMessage msg) {
+        // fast-path to save expensive O(n) checks; users can override createMessage
+        // and extends DefaultHttpRequest making implementing HttpResponse:
+        // this is why we cannot use instanceof DefaultHttpRequest here :(
+        if (msg.getClass() == DefaultHttpRequest.class) {
```

Comment:
@normanmaurer 
It's not nice, but it would save the common path (not overridden `createMessage`) to be super fast. without checking the slow path (not implemented `HttoResponse` type check)